### PR TITLE
Fix layer norm batch norm args

### PR DESCRIPTION
**Pull request by phlrain**
### PR types
Bug fixes

### PR changes
OPs

### Describe
fix layer norm batch norm args order

### CODE REVIEW

File: paddle/phi/infermeta/ternary.cc [hunk 1]
```diff
@@ -209,6 +209,31 @@ void GraphSendRecvInferMeta(const MetaTensor& x,
...
+                        MetaTensor* out,
+                        MetaTensor* mean,
+                        MetaTensor* variance,
+                        MetaConfig config) {
+  auto x_dim = x.dims();
+
+  auto matrix_dim = phi::flatten_to_2d(x_dim, begin_norm_axis);
+  int left = static_cast<int>(matrix_dim[0]);
+
+  out->set_dims(x_dim);
```
Comment (chenwhql) [inline paddle/phi/infermeta/ternary.cc:227]: 原先的检查都没迁过来，会不会有风险

File: paddle/phi/infermeta/ternary.cc [hunk 2]
```diff
@@ -209,6 +209,31 @@ void GraphSendRecvInferMeta(const MetaTensor& x,
...
+  if (mean) {
+    mean->set_dims({left});
+  }
+  if (variance) {
+    variance->set_dims({left});
```
Comment (chenwhql) [inline paddle/phi/infermeta/ternary.cc:232]: 好像都缺set_dtype

File: paddle/phi/infermeta/backward.cc [hunk 3]
```diff
@@ -142,6 +142,23 @@ void GeneralTernaryGradInferMeta(const MetaTensor& x,
...
+    dx->share_meta(x);
+  }
+  if (dy && (y.get_ptr() != nullptr)) {
+    dy->share_meta(*y.get_ptr());
+  }
+  if (dz && (z.get_ptr() != nullptr)) {
+    dz->share_meta(*z.get_ptr());
+  }
+}
+
```
Comment (zyfncg) [inline paddle/phi/infermeta/backward.cc:161]: 函数放置位置按字母序再调整下？

File: paddle/phi/infermeta/unary.cc [hunk 4]
```diff
@@ -360,7 +361,15 @@ void DiagonalInferMeta(const MetaTensor& input,
   out->set_dims(phi::make_ddim(out_dims));
 }
 
-void DropoutInferMeta(const MetaTensor& x, MetaTensor* out, MetaTensor* mask) {
+void DropoutInferMeta(const MetaTensor& x,
+                      paddle::optional<const MetaTensor&> seed_tensor,
```
Comment (zyfncg) [inline paddle/phi/infermeta/unary.cc:365]: 这个需要放在binary文件下了

File: paddle/phi/infermeta/unary.cc [hunk 5]
```diff
@@ -1738,6 +1747,51 @@ void SizeInferMeta(const MetaTensor& input, MetaTensor* out) {
...
+      in_dims, axes, starts, ends, nullptr, &infer_flags);
+  if (config.is_runtime) {
+    out_dims = phi::funcs::GetDecreasedDims<int64_t>(
+        slice_dims, decrease_axis, &infer_flags);
+  } else {
+    out_dims = phi::funcs::GetDecreasedDims<int64_t>(
+        slice_dims, decrease_axis, nullptr);
+  }
+
+  out->set_dims(out_dims);
```
Comment (zyfncg) [inline paddle/phi/infermeta/unary.cc:1789]: dtype设置

File: paddle/phi/kernels/gpu/batch_norm_kernel.cu [hunk 6]
```diff
@@ -466,6 +466,7 @@ void BatchNormKernel(const Context &ctx,
       // backward. Thus this tensor shouldn't be temp.
       // auto *reserve_space = ctx.Output<Tensor>("ReserveSpace");
       if (reserve_space == nullptr) {
+        LOG(ERROR) << "is empty ";
```
Comment (zyfncg) [inline paddle/phi/kernels/gpu/batch_norm_kernel.cu:469]: LOG(ERROR)是调试信息吗？
